checker/match.go

```diff
@@ -193,6 +193,42 @@ func matchHTTPMethods(methods []string, reqMethod string) bool {
 	return false
 }
 
+func matchHTTPPaths(paths []*proto.HTTPMatchPathMatchTypes, reqPath string) bool {
+	log.WithFields(log.Fields{
+		"paths":   paths,
+		"reqPath": reqPath,
+	}).Debug("Matching HTTP Paths")
+	if len(paths) == 0 {
+		log.Debug("Rule has 0 HTTP Paths, matched.")
+		return true
+	}
+	// Accept only valid paths
+	if !strings.HasPrefix(reqPath, "/") {
+		log.Debug("reqPath invalid. (without `/`)")
+		return false
+	}
```

Comment:
I think we need to be louder about the fact that the proxy is sending us invalid requests.  My advice is to `panic` here and recover from it in `Check()` at the top level, logging an Error and maybe returning something like `INVALID_ARGUMENT`.  Matt Klein's blog has some nice detail about how to do this: https://medium.com/@mattklein123/exceptional-go-1dd1488a6a47

(I sort of wish I had done this with the SPIFFE ID parsing, so that we could keep doing that lazily, so maybe one day I'll get back to it.)

Reply:
@spikecurtis Good suggestion. I've done this markup and also addressed rest of the comments.
When u get a chance this is ready for review again.